.travis/run_all_tests.sh

```diff
@@ -2,6 +2,41 @@
 source .travis/utils.sh
 STATUS_CODE=0 # reset
 
+set -ex
+
+####################
+fold_start 'setup_python'
```

Comment:
Maybe this should be moved to init part (before script in Travis) because run_all_tests should be running tests only no setup here. Also it would be nic to enclose all those commands in 1 nice bash script (less code here)

Reply:
Also If you move to separate script you can just call `exit 1` etc

Reply:
Yep - I can move to a separate script, but it has to be `source`d because it's modifying PATH.

Reply:
Yeah and this is why I mean to run it directly from .travis.yml